src/__tests__/index.node.js

```
@@ -112,52 +130,368 @@ test('/load-time caching', async t => {
-          console.log(
-            '[TEST] cached after first load:',
-            '#',
-            keys.map(key => key.url).join(',')
+    await logCachedURLs(page, '[TEST] cached after first load:');
+    await logCacheDates(page, '[TEST] cache dates after first load:');
+
+    // Capture requests during 2nd load.
+    const allRequests = new Map();
+
+    page.on('request', req => {
+      allRequests.set(req.url(), req);
+    });
+
+    // wait one second between loads because puppteer and/or cache API appears to round to nearest second
+    await page.waitFor(1000);
+
+    // SECOND LOAD
+    await page.reload({waitUntil: 'domcontentloaded'});
+    controller = await page.evaluate('navigator.serviceWorker.controller');
+    t.ok(
+      controller,
+      'second page load: page has an existing active service worker'
+    );
+
+    await logCachedURLs(page, '[TEST] cached after second load:');
+    await logCacheDates(page, '[TEST] cache dates after second load:');
+
+    t.ok(
+      Array.from(allRequests.values())
+        .filter(req =>
+          cacheablePaths.find(
+            path =>
+              `${hostname}${port}${path}` === req.url() ||
+              req.url() === `${hostname}${port}/` // html
           )
         )
+        .every(req => req.response() && req.response().fromServiceWorker()),
+      'all cacheable resources are fetched from service worker'
+    );
+
+    t.ok(
+      Array.from(allRequests.values())
+        .filter(
+          req =>
+            !cacheablePaths.find(
+              path => `${hostname}${port}${path}` === req.url()
+            )
+        )
+        .every(
+          req =>
+            req.resourceType() === 'document' || // html always processed by SW
+            !req.response() ||
+            !req.response().fromServiceWorker()
+        ),
+      'all non-cacheable resources are not fetched from service worker'
+    );
+
+    await browser.close();
+  } finally {
+    proc.kill();
+    t.end();
```

I'm wondering if we're swallowing exceptions in tests here? Did you try adding a `throw new Error()` inside the try statement to make sure it still fails?